src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -99,30 +53,41 @@ internal TagHelperAttributeDescriptor([NotNull] string name, [NotNull] PropertyI
         /// to allow empty values for HTML attributes matching this <see cref="TagHelperAttributeDescriptor"/>. If
         /// <c>false</c> empty values for such matching attributes lead to errors.
         /// </value>
-        public bool IsStringProperty { get; }
+        public bool IsStringProperty { get; set; }
```

Comment:
Do not allow this to be publicly set. At most `internal set` for testing purposes; `private set` if we don't have any tests like that.